### DISCUSSION TITLE
How to add multiple languages using Lighthouse GraphQL.

### DISCUSSION THREAD

**Discussion by M7MMAD-OMAR** [nuwave/lighthouse#2551] [Q&A]
Hello everyone,
I have a Laravel project and I want to add multiple languages to it. I have used the library `spatie/laravel-translatable` and set the default language of the project to Turkish. It works and takes the Turkish language. However, my computer's language is English, and I want my project to take the computer's language and return the data in the appropriate language. How can I handle this using Lighthouse? It takes the translation but does not recognize my computer's language.

Even though I added the middleware in the `Kernel` class,
This is the middleware.

```php
<?php

namespace App\Http\Middleware;

use Closure;
use Illuminate\Http\Request;
use Modules\AccountManagement\Enums\UserProfile\LanguageEnum;

class LangMiddleware
{
    /**
     * Handle an incoming request.
     *
     * @param \Closure(\Illuminate\Http\Request): (\Symfony\Component\HttpFoundation\Response) $next
     */
    protected const array ALLOWED_LOCALIZATIONS = ['ar', 'en', 'tr'];

    public function handle(Request $request, Closure $next)
    {
        $localization = $request->header('Accept-Language');
        $localization = in_array($localization, self::ALLOWED_LOCALIZATIONS, true) ? $localization : 'ar';
        app()->setLocale($localization);
        return $next($request);
    }
}
```

**Comment by pyrou** [comment]
Hi,

First things to know, the `Accept-Language` header would not contains just `ar` or `en` but something like `ar, en-GB;q=0.8, en;q=0.7`

Luckily, Laravel got you covered on parsing these priority, and will select the available languages that fit the best. 

Here is the implementation :

```php
$locale = $request->getPreferredLanguage(self::ALLOWED_LOCALIZATIONS);
app()->setLocale($locale);
```
